test/rules/always_put_control_body_on_new_line.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N always_put_control_body_on_new_line`
+
+testIfElse() {
```

Comment:
I thought it should too, but looking at the code, I don't think it will. Because we compare the line containing the right paren for the condition with the line containing the semicolon, I think it will decide that the whole statement is on a separate line. I think you need to use the first character of the `node` and have special handling for blocks, such as testing the location of the first character of the first statement in the block. (I'd ignore empty blocks, because that seems like something that should be covered by a different lint.) It would be good to add a test of that form.